### ISSUE TITLE
Incompatible tool - fubectl

### ISSUE THREAD

**Issue by tp-woven** [warpdotdev/Warp#3263]
### Discord username (optional)

tal8036

### Describe the bug

TL;DR: Having [fubectl](https://github.com/kubermatic/fubectl) "installed" causes Warp to hang when opening a shell.

I have the fubectl init code in `$ZSH_CUSTOM/fubectl.zsh`, and commenting it out "solves" the issue:

```shell
[ -f ${HOME}/bin/fubectl.source ] && source ${HOME}/bin/fubectl.source
```

I am currently using [this version](https://github.com/kubermatic/fubectl/blob/3950b1e3482aff14e810af2459a6e213dd89364d/fubectl.source) of fubectl (although it didn't work with an older version either).

I'm not sure if it's related, but I am also using OMZ and p10k (latest version, as well as latest version of Warp).

### To reproduce

1. Install [fubectl](https://github.com/kubermatic/fubectl/tree/3950b1e3482aff14e810af2459a6e213dd89364d#installation)
2. Open a new shell

=> Gets stuck on "Starting zsh..."

### Operating system

MacOS

### Operating system and version

Ventura 13.4.1

### Shell Version

zsh 5.9 (x86_64-apple-darwin22.0)

### Current Warp version

v0.2023.06.20.08.04.stable_03

### Regression

No, this bug or issue has existed throughout my experience using Warp

### Does this block you from using Warp daily?

No

### Is this a Warp specific issue? (i.e. does it happen in Terminal, iTerm, Kitty, etc.)

Yes, this I confirmed this only happens in Warp, not other terminals.

### Warp Internal (ignore): linear-label:b8107fdf-ba31-488d-b103-d271c89cac3e

None

**Comment by dannyneira** [comment]
Thanks for letting us know @tp-woven I was able to reproduce the issue with `fubectl`. As a workaround to the issue, you can disable it just for Warp by putting the code in the following conditionals for the time being.

```sh
# Bash and Zsh
if [[ $TERM_PROGRAM != "WarpTerminal" ]]; then
[ -f ${HOME}/bin/fubectl.source ] && source ${HOME}/bin/fubectl.source
fi
```

I've also added it to our list of incompatible tools [here](https://docs.warp.dev/help/known-issues#list-of-incompatible-tools), but we'll post any updates/changes in support on this thread.

Please add a :+1: to the original post at the top or comment with your details to signal that you're also having the same issue, and then hit the subscribe button if you'd like to be notified.

**Comment by tp-woven** [comment]
Thanks @dannyneira! Yes, I did add the `TERM_PROGRAM` check and it resolves the issue. 👍 

BTW, going over the list of unsupported tools again, I wonder if this is related to fzf, which fubectl uses quite heavily... 🤔
